internal/controller/deployment_controller.go

```diff
@@ -87,6 +89,63 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.Update(ctx, l, deployment)
 }
 
+func (r *DeploymentReconciler) setStatusFromClusterStatus(ctx context.Context, l logr.Logger, deployment *hmc.Deployment) (bool, error) {
+	dc, err := dynamic.NewForConfig(r.Config)
+	if err != nil {
+		return true, fmt.Errorf("failed to create client configuration: %v", err)
+	}
+
+	resourceId := schema.GroupVersionResource{
+		Group:    "cluster.x-k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusters",
+	}
+
+	list, err := dc.Resource(resourceId).Namespace(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		l.Info("Cluster not found, ignoring since object must be deleted or not yet created")
+		return true, nil
+	}
+
+	if err != nil {
+		return true, fmt.Errorf("failed to get cluster information for deployment: %s : %v", deployment.Name, err)
+	}
+
+	requeue := true
+	if statusField, ok := list.Object["status"]; ok {
```

Comment:
You don't need to do this, the API returns `unstructured.Unstructured` types and there's an entire library to help you make these conversions and get this content much much easier:

```go
	conditions, found, err := unstructured.NestedMap(list.Object, "status", "conditions")
	if err != nil {
		return true, fmt.Errorf("failed to get cluster information for deployment: %s: %w", deployment.Name, err)
	}
	if !found {
		return true, fmt.Errorf("failed to get cluster information for deployment: %s: status.conditions not found", deployment.Name)
	}

	var conditionsList []metav1.Condition

	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(conditions, &conditionsList); err != nil {
		return true, fmt.Errorf("failed to convert unstructured conditions to []metav1.Condition for deployment: %s: %w", deployment.Name, err)
	}

	allConditionsComplete := true
	for _, condition := range conditionsList {
		if condition.Status != "True" {
			allConditionsComplete = false
		}

		if condition.Reason == "" && condition.Status == "True" {
			condition.Reason = "Succeeded"
		}
		apimeta.SetStatusCondition(deployment.GetConditions(), condition)
	}
	requeue := !allConditionsComplete

	return requeue, nil
```

Reply:
Kind of works thanks! Turns out the conditions field is a list of interfaces and not maps though so I had to change it a bit.